src/java.base/share/classes/jdk/internal/reflect/NativeConstructorAccessorImpl.java

```diff
@@ -51,14 +57,23 @@ public Object newInstance(Object[] args)
         // be found from the generated bytecode.
         if (++numInvocations > ReflectionFactory.inflationThreshold()
                 && !c.getDeclaringClass().isHidden()
-                && !ReflectUtil.isVMAnonymousClass(c.getDeclaringClass())) {
-            ConstructorAccessorImpl acc = (ConstructorAccessorImpl)
-                new MethodAccessorGenerator().
-                    generateConstructor(c.getDeclaringClass(),
-                                        c.getParameterTypes(),
-                                        c.getExceptionTypes(),
-                                        c.getModifiers());
-            parent.setDelegate(acc);
+                && !ReflectUtil.isVMAnonymousClass(c.getDeclaringClass())
+                && accessorGenerated == false
+                && unsafe.compareAndSetBoolean(this, accessorGeneratedaOffset, false, true)) {
```

Comment:
`compareAndSetBoolean` gets us into the awkward territory of sub-word CASes. (Click through to `Unsafe.compareAndExchangeByte` to see what I am talking about). It is probably not relevant here, though. Still, might be as good to use `int` field for generated flag.

Reply:
use int type now